source/FreeRTOS_TCP_Transmission_IPV4.c

```diff
@@ -502,11 +502,22 @@
                 /* coverity[misra_c_2012_rule_11_3_violation] */
                 TCPPacket_t * pxTCPPacket = ( ( TCPPacket_t * ) pxNetworkBuffer->pucEthernetBuffer );
                 const uint32_t ulSendLength =
-                    ( uxIPHeaderSizePacket( pxNetworkBuffer ) + ipSIZE_OF_TCP_HEADER ); /* Plus 0 options. */
+                    ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_TCP_HEADER; /* Plus 0 options. */
 
+                uint8_t ucFlagsReceived = pxTCPPacket->xTCPHeader.ucTCPFlags;
                 pxTCPPacket->xTCPHeader.ucTCPFlags = ucTCPFlags;
                 pxTCPPacket->xTCPHeader.ucTCPOffset = ( ipSIZE_OF_TCP_HEADER ) << 2;
 
+                if( ( ucFlagsReceived & tcpTCP_FLAG_SYN ) != 0U )
```

Comment:
Hi Hein, can you explain a little more on this, for which case do we need this

Reply:
See #724 which is already merged.
When replying a SYN packet, the ACK should always be equal to SEQ+1. And as `ulSequenceNumber` is stored as network-endian, it must be converted twice:

~~~c
    uint32_t ulSequenceNumber = FreeRTOS_ntohl( pxTCPPacket->xTCPHeader.ulSequenceNumber );
    ulSequenceNumber++;
    pxTCPPacket->xTCPHeader.ulSequenceNumber = FreeRTOS_htonl( ulSequenceNumber );
~~~
I made the same change to `prvTCPSendSpecialPktHelper_IPV6()`

Reply:
Thanks for clarifying